src/consensus/consensus.h

```diff
@@ -6,10 +6,26 @@
 #ifndef BITCOIN_CONSENSUS_CONSENSUS_H
 #define BITCOIN_CONSENSUS_CONSENSUS_H
 
+#include <stdint.h>
+
+static const uint64_t BIP102_FORK_TIME = 1462406400; // May 5 2016, midnight UTC
+
 /** The maximum allowed size for a serialized block, in bytes (network rule) */
-static const unsigned int MAX_BLOCK_SIZE = 1000000;
+inline unsigned int MaxBlockSize(uint64_t nTime) {
+    if (nTime < BIP102_FORK_TIME)
+        return 1000*1000;
+
+    return (2*1000*1000);
```

Comment:
Should also be in Consensus::Params, and the current limit as well. Also why 1000*1000 instead of 1000000 like it was?